wgengine/magicsock/magicsock.go

```diff
@@ -2598,6 +2618,7 @@ func (c *Conn) SetNetworkMap(nm *netmap.NetworkMap) {
 
 		ep := &endpoint{
 			c:                 c,
+			debugUpdates:      ringbuffer.New[EndpointChange](100),
```

Comment:
could this be too much for a large tailnet?

Reply:
I just pushed a change that limits the total number of ring buffer entries, counted by dividing `len(nm.Peers)`. I added a new envknob type (`RegisterInt`) for this.